tutorials/preprocessing/plot_35_artifact_correction_regression.py

```diff
@@ -0,0 +1,142 @@
+                  verbose='error')  # ignore warnings about spatial colors
+fig = epochs.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('Auditory epochs')
+fig.tight_layout()
+
+epochs_no_ave = epochs.copy().subtract_evoked()
+_, betas = mne.preprocessing.regress_artifact(epochs_no_ave)
+epochs_clean, _ = mne.preprocessing.regress_artifact(epochs, betas=betas)
+fig = epochs_clean.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('Auditory epochs, EOG regressed')
+fig.tight_layout()
+del epochs, epochs_clean
+
+###############################################################################
+# The effect is subtle in these evoked data. You can see that a bump toward
+# the end of the window has had its amplitude decreased.
+#
+# The effect is even clearer if we create epochs around our (autodetected) EOG
+# events and plot the average across those epochs:
+
+eog_epochs = mne.preprocessing.create_eog_epochs(raw, decim=decim)
+eog_epochs.apply_baseline((None, None))
+order = np.concatenate([  # plotting order: EOG+ECG first, then MEG
+    mne.pick_types(raw.info, meg=False, eog=True, ecg=True),
+    mne.pick_types(raw.info, meg=True)])
+raw_kwargs = dict(order=order, duration=25, n_channels=40)
+raw.plot(events=eog_epochs.events, **raw_kwargs)
+fig = eog_epochs.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('EOG epochs')
+fig.tight_layout()
+
+###############################################################################
+# And then clean those data:
+
+raw_clean, _ = mne.preprocessing.regress_artifact(raw, betas=betas)
+raw_clean.plot(events=eog_epochs.events, **raw_kwargs)
+eog_epochs_clean, _ = mne.preprocessing.regress_artifact(eog_epochs,
+                                                         betas=betas)
+fig = eog_epochs_clean.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('EOG epochs, EOG regressed')
+fig.tight_layout()
+del eog_epochs, eog_epochs_clean, raw_clean  # save memory
+
+###############################################################################
+# ECG
+# ~~~
+# We can follow the same process with ECG:
+
+ecg_epochs = mne.preprocessing.create_ecg_epochs(raw, decim=decim)
+ecg_epochs.apply_baseline((None, None))
+raw.plot(events=ecg_epochs.events, **raw_kwargs)
+fig = ecg_epochs.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('ECG epochs')
+fig.tight_layout()
+
+###############################################################################
+# But it does not remove as much of the signal, even from the reference
+# epochs, likely because the ECG signal is a rotating dipole, and the ECG
+# electrode time waveform does not capture the same effective time waveform
+# that each channel does:
+
+# Here we operate in place to save memory
+mne.preprocessing.regress_artifact(
+    ecg_epochs, picks_artifact='ecg', copy=False)
+fig = ecg_epochs.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('ECG epochs, ECG regressed')
+fig.tight_layout()
```

Comment:
same feedback as for EOG section: put raw plots side by side, and (if included) put ECG epochs side by side and call attention to y-axis limits.

Reply:
Overall, this example doesn't impress me much - as stated in the introduction, this approach does not work well with ECG artifacts. I've never seen anyone use the regression approach for ECG artifacts, so why do we show it here? I would remove this example completely.